lib/ansible/modules/crypto/luks_device.py

```diff
@@ -376,17 +432,32 @@ def run_luks_remove(self, device):
             raise ValueError('Error while wiping luks container %s: %s'
                              % (device, result[STDERR]))
 
-    def run_luks_add_key(self, device, keyfile, new_keyfile):
-        ''' Add new key to given 'device'; authentication done using 'keyfile'
-            Raises ValueError when command fails
+    def run_luks_add_key(self, device, keyfile, passphrase, new_keyfile,
+                         new_passphrase):
+        ''' Add new key from a keyfile or passphrase to given 'device';
+            authentication done using 'keyfile' or 'passphrase'.
+            Raises ValueError when command fails.
         '''
-        result = self._run_command([self._cryptsetup_bin, 'luksAddKey', device,
-                                    new_keyfile, '--key-file', keyfile])
+        data = []
+        args = [self._cryptsetup_bin, 'luksAddKey', device]
+
+        if keyfile:
+            args.extend(['--key-file', keyfile])
+        else:
+            data.append(passphrase)
+
+        if new_keyfile:
+            args.append(new_keyfile)
+        else:
+            data.extend([new_passphrase, new_passphrase])
+
+        result = self._run_command(args, data='\n'.join(data) or None)
         if result[RETURN_CODE] != 0:
             raise ValueError('Error while adding new LUKS key to %s: %s'
```

Comment:
```suggestion
            raise ValueError('Error while adding new LUKS keyslot to %s: %s'
```
While you're at it ;)